pkg/storage/formattedstore/formattedstore.go

```diff
@@ -112,8 +115,29 @@ func (s *formatStore) Get(k string) ([]byte, error) {
 	return value, nil
 }
 
-func (s *formatStore) Iterator(startKey, endKey string) StoreIterator {
-	return s.store.Iterator(startKey, endKey)
+func (s *formatStore) Iterator(startKey, endKey string) storage.StoreIterator {
+	formattedIterator := s.store.Iterator(startKey, endKey)
+
+	errFormattedIterator := formattedIterator.Error()
+	if errFormattedIterator != nil {
+		return mem.NewMemIterator(nil,
+			fmt.Errorf("failed to get iterator from underlying store: %w", errFormattedIterator))
+	}
+
+	var batch [][]string
+
+	for formattedIterator.Next() {
```

Comment:
This will load all values into memory. Would it be better for the MemIterator to hold on the the formattedIterator and retrieve the next value when memIterator.Next() is invoked, and then release the formattedIterator when MemIterator.Release() is called.

Reply:
The code has been updated since you originally commented. With the new way it works, the values must be deformatted in order to apply the startKey-endKey filter, so I don't think it's possible to avoid loading all the values in memory... but let me know if you think it's still possible.